lib/router/handler-info.js

```diff
@@ -24,7 +27,52 @@ function HandlerInfo(_props) {
 
 HandlerInfo.prototype = {
   name: null,
-  handler: null,
+
+  getHandler: function() {},
+
+  get handler() {
+    if (this.hasOwnProperty('_handler')) {
```

Comment:
Why do we need the has own check here? Is there a scenario where we have inherited a `_handler`?

Would be easier/better?

```js
if (this._handler) { return this._handler; }
```

Also, `_handler` needs to be initialized to `undefined` or `null` in the constructor to avoid shaping issues (which I think would make the `hasOwnProperty` check fail to work since it would have the property but it would be undefined).